### ISSUE TITLE
Netty incorrectly strips `\r`, `\x0b`, and `\x0c` bytes from the ends of header values

### ISSUE THREAD

**Issue by kenballus** [netty/netty#14167]
### Steps to reproduce
1. Start a Netty HTTP/1.1 server that echos the parsed components of the request, such as [this one](https://raw.githubusercontent.com/narfindustries/http-garden/main/images/netty/GardenServer.java).
2. Send it a request with a header value suffixed with any number of `\r`, `\x0b`, and `\x0c` bytes in any order, and extract the header value from the response:
```sh
(printf 'GET / HTTP/1.1\r\nHost: whatever\r\nTest-Key: test-value\x0c\x0b\r\x0b\x0c\r\n\r\n'; sleep 1) \
    | ncat localhost 80 \
    | grep '"headers"' \
    | jq '.["headers"][1][1]' \
    | xargs echo \
    | base64 -d \
    | xxd
```
4. Observe that the extracted header value is missing the aforementioned bytes:
```
00000000: 7465 7374 2d76 616c 7565                 test-value
```

### Expected behavior
The RFCs permit two behaviors:
1. Reject the request, since these characters are not permitted within header values.
    - AIOHTTP, Apache, Deno, FastHTTP, Go net/http, H2O, HAProxy, Hyper, Hypercorn, Jetty, Libevent, Lighttpd, Mongoose, Netty, Nginx, Node.js, Passenger, Puma, Tomcat, Uvicorn, Waitress, and WEBrick do this.
2. Translate the `\x0d` into SP, then process the request, appropriately stripping SP and HTAB bytes (including those just created) and allowing `\x0b` and `\x0c` to stay in the value.
    - Libsoup, LiteSpeed, and Twisted do this.

### Actual behavior
Netty strips the aformentioned bytes.

### Netty version
4.1 @ 967be496c6d9733062883344faec538b1854d2a8

### JVM version (e.g. `java -version`)
```
openjdk version "23-ea" 2024-09-17
OpenJDK Runtime Environment (build 23-ea+20-Debian-1)
OpenJDK 64-Bit Server VM (build 23-ea+20-Debian-1, mixed mode, sharing)
```

### OS version (e.g. `uname -a`)
```
Linux 416cc96d1b8b 6.9.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000 x86_64 GNU/Linux
```

**Comment by normanmaurer** [comment]
@bryce-anderson @idelpivnitskiy @vietj @violetagg thoughts on if we should do 1) or better 2) ?

**Comment by idelpivnitskiy** [comment]
In ServiceTalk we do (1). Netty's decoder was always more lenient to RFC for backward compatibility with older systems. However, taking into account a long list of systems that follow (1), I'm leaning towards (1) as well.

**Comment by normanmaurer** [comment]
@idelpivnitskiy yeah same

**Comment by violetagg** [comment]
I would vote for (1)

**Comment by kenballus** [comment]
I also vote for 1, as much as my vote counts :)

The whole translating to SP thing is pretty brittle. For example, LiteSpeed proxy + Gunicorn is currently vulnerable to request smuggling because Gunicorn translates bare LF to SP and LiteSpeed forwards bare LF without normalizing to CRLF.

**Comment by normanmaurer** [comment]
@kenballus want to do a PR?

**Comment by vietj** [comment]
just saw the ping, option 1 looks good to me

**Comment by kenballus** [comment]
Confirmed; this is fixed by #14178.
